src/Sulu/Component/Webspace/Analyzer/RequestAnalyzer.php

```diff
@@ -45,11 +51,25 @@ public function analyze(Request $request)
     }
 
     /**
+     * Initilizes the request attributes lazy.
+     *
+     * @return RequestAttributes
+     */
+    protected function getAttributes()
+    {
+        if (!$this->attributes) {
```

Comment:
The caching of these values is distributed over three functions (`getAttributes`, `createAttributes` and `analyze`). I would think about combining that into a single method. This way, if the `analyze` method is called directly, the attributes  are calculated again, although it is not necessary. In an ideal world I'd say we remove the `analyze` method (sadly it is part of our BC promise) and create attributes. For now I would combine these three methods into the protected method  `getAttribute`, and handle all of this stuff there. In addition I would deprecate the `analyze` method, and let it simply call the `getAttribute` method, so that everything gets initialized.